google/cloud/spanner/retry_policy.h

```diff
@@ -28,8 +28,24 @@ namespace internal {
 /// Define the gRPC status code semantics for retrying requests.
 struct SafeGrpcRetry {
   static inline bool IsTransientFailure(google::cloud::StatusCode code) {
-    return code == StatusCode::kAborted || code == StatusCode::kUnavailable ||
-           code == StatusCode::kDeadlineExceeded;
+    return code == StatusCode::kUnavailable ||
+           code == StatusCode::kResourceExhausted;
+  }
+  static inline bool IsOk(google::cloud::Status const& status) {
+    return status.ok();
+  }
+  static inline bool IsTransientFailure(google::cloud::Status const& status) {
+    return IsTransientFailure(status.code());
+  }
+  static inline bool IsPermanentFailure(google::cloud::Status const& status) {
+    return !IsOk(status) && !IsTransientFailure(status);
+  }
+};
+
+/// Define the gRPC status code semantics for rerunning transactions.
+struct SafeTransactionReRun {
```

Comment:
Here and throughout, treat "rerun" as a single word, just like "retry" (see Merriam-Webster for example).

Reply:
Done.